src/Build/Definition/ToolsetConfigurationReader.cs

```diff
@@ -250,6 +257,18 @@ protected override IEnumerable<ToolsetPropertyDefinition> GetSubToolsetPropertyD
         /// Unit tests wish to avoid reading (nunit.exe) application configuration file.
         /// </summary>
         private static Configuration ReadApplicationConfiguration()
+        {
+            if (Environment.GetEnvironmentVariable("MSBUILDCACHETOOLSETCONFIGURATION") != "0")
```

Comment:
I am not completely sure, when to use changewave vs escape-hatches. I am inclining to use changewave when msbuild change its behavior and when probability of regression is considerable. 
The fact that user requires previous behavior of msbuild does not necessary means he wants to reject perf optimizations as well.
Though I might be looking at it wrong. Please let me know if you prefer to use changewave here.

Reply:
For a thing like this where we hope it is never needed, I think a changewave is best since they'll "naturally" expire as we follow the policy. I tend to favor explicit switches (usually in addition to a changewave) only if we expect to need the configurability "forever", like in #6560.

@BenVillalobos or @marcpopMSFT might have more detailed or different thoughts though.

Reply:
Makes sense. I have moved it under 17_0 change wave.

Reply:
To add onto rainer's thoughts, if there were backlash such that we know a permanent workaround is needed then we can always add the secondary/permanent escape hatch later. Perhaps as part of the wave "rotating out" when we've gathered more real world data.